### PR TITLE
Modify at VertexAttribute

### PR DESCRIPTION
**Pull request by leeho0108**
Like say in #3933 ,just make `private` to `protected`

### CODE REVIEW

**Comment by xoppa** [comment]
Since you are making these constructors part of the api, please add the necessary javadocs. Including a warning.

**Comment by xoppa** [comment]
Whats the status of this @leeho0108?

**Comment by leeho0108** [comment]
@xoppa 
Sorry, I almost forgot this ,because I am recently to busy.
I add some docs to this two protected constructor in my fork
https://raw.githubusercontent.com/leeho0108/libgdx/master/gdx/src/com/badlogic/gdx/graphics/VertexAttribute.java
But my English is too bad...
So, It will be great if you could look at it first and correct any words. 
Thanks

**Comment by badlogic** [comment]
@xoppa can you fix up the docs and merge? Looks OK to me.

File: gdx/src/com/badlogic/gdx/graphics/VertexAttribute.java [hunk 1]
```diff
@@ -16,8 +16,13 @@
 
 package com.badlogic.gdx.graphics;
 
+import java.nio.FloatBuffer;
+
+import javafx.beans.binding.FloatBinding;
```
Comment (Darkyenus) [inline gdx/src/com/badlogic/gdx/graphics/VertexAttribute.java]: What's up with this?

**Comment by leeho0108** [comment]
@Darkyenus  It's my mistake,I had remove this useless import.

**Comment by cypherdare** [comment]
@xoppa 
`usage` is an obsolete parameter since we no longer support fixed pipeline. I think it can remain in the constructor for backward compatibility, but there's no reason for storing its value or using it in `hashCode()`. And `getKey()` could use `type` instead.

Then, all cases (such as in VertexAttributes.calculateOffsets()) of `if (attribute.usage == VertexAttributes.Usage.ColorPacked)` could be replaced with `if (attribute.type == GL20.GL_UNSIGNED_BYTE)`, and these protected constructors would be safer to use. They could even be public. And we then we could simply copy-paste the docs from the other constructors.

**Comment by cypherdare** [comment]
@xoppa Hold on, I see that you're using Usage for additional info in the 3D api. I guess it should be kept, but docs updated since they mention in multiple places that they're only for fixed function. I still think using `if (attribute.usage == VertexAttributes.Usage.ColorPacked)` anywhere should be changed.

**Comment by xoppa** [comment]
Thanks @leeho0108! Sorry for the delay.

Care to send a PR for that @cypherdare?

**Comment by cypherdare** [comment]
@xoppa, take a look at #4395. It could use your input.

**Pull request merged**